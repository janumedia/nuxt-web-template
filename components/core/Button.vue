<template>
    <button :class="className" :type="submit !== undefined ? 'submit' : ''">
        <nuxt-link v-if="url && url.indexOf('http') == -1" :title="label" :to="url">{{ label }}</nuxt-link>
        <a v-else-if="url" :href="url" target="_blank" :title="label" rel="noopener">{{ label }}</a>
        <span v-else>{{ label }}</span>
    </button>
</template>
<script>
export default {
    props: ["primary", "submit", "label", "url"],
    computed: {
        className() {
            let str = this.primary !== undefined ? 'button--primary' : '';
            return str + (this.submit !== undefined ? ' button--submit' : '');
        }
    }
}
</script>
<style lang="scss" scoped>
@import "@/assets/css/base.scss";
button {
    border: 1px solid #ccc;;
    border-radius: 0;
    -webkit-border-radius: 0;
    -moz-border-radius: 0;    
    
    text-transform: uppercase;
    letter-spacing: 0.04em;
    display:inline-block;
    padding: 0;
    vertical-align:middle;
    text-decoration:none;
    font-size: 0.8em;
    color:#807e7e;
    background-color:inherit;
    text-align:center;
    cursor:pointer;
    white-space:nowrap;
    @include transition-duration(0.4s);

    &:focus {
        outline: none;
    }
    &:hover {
        color: $bgColor;
        background-color:#ccc;
    }
    & span,
    & a {
        display:block;
        color:#807e7e;
        padding:8px 16px;
        @include transition-duration(0.4s);
        cursor: pointer;
    }
    & > a:hover {
        color: $bgColor;
        background-color:#ccc;
    }
}

.button--primary {
    border: 1px solid $mainColor;
    color: $mainColor;
    padding:0px 9px;
    background-color: $mainColor;
    margin-bottom: 16px;
    span,
    a {
        color: $bgColor;
        &:hover {
            background-color:$errorColor;
        }
    }
    &:hover {
        background-color:$errorColor;
    }
}

.button--submit {
    padding:8px 40px;
}

@media only screen and (max-width:768px) {
    .button--submit {
        width: 100%;
    }
}

</style>
