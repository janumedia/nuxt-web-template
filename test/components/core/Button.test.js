import { shallowMount } from '@vue/test-utils'
import NuxtLink from '@/.nuxt/components/nuxt-link'
import Button from '@/components/core/Button'

describe('Button.vue', () => {
    let wrapper, btnLabel = 'button label';
    beforeEach(() => {
        Button.components = Button.components || {}
        Button.components.NuxtLink = NuxtLink
    });
    it('should render button with router', () => {
        wrapper = shallowMount(Button, {
            propsData: {
                url: 'router/link',
                label: btnLabel
            }
        })
        expect(wrapper.html()).toMatchSnapshot();
    });
    it('should render button with a href', () => {
        wrapper = shallowMount(Button, {
            propsData: {
                url: 'http://domain.com',
                label: btnLabel
            }
        })
        expect(wrapper.html()).toMatchSnapshot();
    });
    it('should render button with primary theme', () => {
        wrapper = shallowMount(Button, {
            propsData: {
                url: 'http://domain.com',
                label: btnLabel,
                primary: true
            }
        })
        expect(wrapper.html()).toMatchSnapshot();
    })
    it('should render button type submit', () => {
        wrapper = shallowMount(Button, {
            propsData: {
                label: btnLabel,
                submit: true
            }
        })
        expect(wrapper.html()).toMatchSnapshot();
    })
})