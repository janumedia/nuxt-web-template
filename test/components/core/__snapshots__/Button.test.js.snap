// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue should render button type submit 1`] = `<button type="submit" class=" button--submit"><span>button label</span></button>`;

exports[`Button.vue should render button with a href 1`] = `
<button type="" class="">
  <a href="http://domain.com" target="_blank" title="button label" rel="noopener">button label</a>
</button>
`;

exports[`Button.vue should render button with primary theme 1`] = `
<button type="" class="button--primary">
  <a href="http://domain.com" target="_blank" title="button label" rel="noopener">button label</a>
</button>
`;

exports[`Button.vue should render button with router 1`] = `
<button type="" class="">
  <nuxtlink-stub title="button label" to="router/link">button label</nuxtlink-stub>
</button>
`;
